### PR TITLE
Only show size for regular files

### LINKED ISSUE #531: Meaningless size label shown for folders
Folders gets a "0 B" added under them, which isn't at all helpful.  I assume that's the `st_size` which is reported as 0 by btrfs.  Other filesystems might report something different there, but it's never going to be interesting to the user.

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
This avoids showing a meaningless '0 B' (or misleading arbitrary other size) for folders.  Other special files also lack a meaningful size, so don't show them either.

Symlinks are a bit of a tricky case here: we currently show the size of the link (which is the length of the link target in bytes).  We could show the size of the target, but this is a bit odd.  Let's show nothing.

Fixes #531

### CODE REVIEW

**Comment by martinpitt** [comment]
That should come with integration tests.

File: test/check-application [hunk 1]
```diff
@@ -1292,6 +1294,96 @@ class TestFiles(testlib.MachineCase):
...
+
+        self.enter_files()
+        b.click("button[aria-label='Display as a list']")
+
+        # For each file we created, assert various things about how we expect
+        # it to be displayed.
+        for name in files:
+            selector = f"[data-item='{name}']"
+            classes = b.attr(selector, 'class').split()
+            assert b.text(f'{selector} .item-name') == name
```
Comment (martinpitt) [inline test/check-application:1352]: THis isn't pytest yet, so `self.assertEqual()` please for useful failures.

File: test/check-application [hunk 2]
```diff
@@ -1292,6 +1294,96 @@ class TestFiles(testlib.MachineCase):
...
+            size = b.text(f'{selector} [data-label="size"]')
+            date = b.text(f'{selector} [data-label="date"]')
+
+            if 'dir' in name:
+                # directories are shown with folder icon, regardless of extension
+                assert 'folder' in classes
```
Comment (martinpitt) [inline test/check-application:1358]: self.assertIn()

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks!

File: test/check-application (hunk 1 above)
Comment (allisonkarlitskaya) [reply test/check-application]: done x4

File: test/check-application (hunk 2 above)
Comment (allisonkarlitskaya) [reply test/check-application]: done x5

**Review by martinpitt — APPROVED** [review]
Cheers!

**Review by jelly — APPROVED** [review]

**Pull request merged**